instance/models/openedx_instance.py

```diff
@@ -220,10 +220,24 @@ def set_active_vm_dns_records(self):
         self.logger.info("Setting DNS records for active app servers...")
         for i, appserver in enumerate(self.get_active_appservers(), 1):
             ip_addr = appserver.server.public_ip
+
             if ip_addr:
                 domain = "vm{index}.{base_domain}".format(index=i, base_domain=self.internal_lms_domain)
                 gandi.api.set_dns_record(domain, type="A", value=ip_addr)
-        # TODO: implement cleaning up DNS addresses that are no longer needed.
+
+        active_appservers = self.get_active_appservers()
+        unused_dns_index = active_appservers.count() + 1
+        domain = "vm{index}.{base_domain}".format(index=unused_dns_index, base_domain=self.internal_lms_domain)
```

Comment:
@pkulkark, this `set_active_vm_dns_records()` method is called from the `make_active()` method [here](https://github.com/open-craft/opencraft/blob/master/instance/models/openedx_appserver.py#L283) which already knows whether the appserver is getting deactivated or not. So if that information is passed to this method via variable named, say `deactivate_an_appserver`, the DNS record removal here can be done only if that is the case. This will help avoid making an additional deletion call every time irrespective of whether the DNS record exists or not.

Reply:
Makes sense. Added a variable.